lte/gateway/c/core/oai/include/map.h

```diff
@@ -0,0 +1,232 @@
+  ** Description: Takes key and valueP as parameters.If the key exists,     **
+  **              corresponding value is returned through the valueP,       **
+  **              else returns error.                                       **
+  **                                                                        **
+  ***************************************************************************/
+  map_rc_t get(const keyT key, valueT* valueP) {
+    if (umap.empty()) {
+      return MAP_EMPTY;
+    }
+    if (valueP == nullptr) {
+      return MAP_BAD_PARAMETER_VALUE;
+    }
+    std::lock_guard<std::mutex> lock(mutex_obj);
+    auto search_result = umap.find(key);
+    if (search_result != umap.end()) {
+      *valueP = search_result->second;
+      return MAP_OK;
+    } else {
+      return MAP_KEY_NOT_EXISTS;
+    }
+  }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    insert()                                                      **
+  **                                                                        **
+  ** Description: Takes key and value as parameters.Inserts the <key,value> **
+  **              pair into the map. If identical key already exists then   **
+  **              returns error.                                            **
+  **                                                                        **
+  ***************************************************************************/
+  map_rc_t insert(const keyT key, const valueT value) {
+    typedef typename std::unordered_map<keyT, valueT>::iterator itr;
+    std::lock_guard<std::mutex> lock(mutex_obj);
+    std::pair<itr, bool> insert_response =
+        umap.insert(std::make_pair(key, value));
+    if (insert_response.second) {
+      return MAP_OK;
+    } else {
+      return MAP_KEY_ALREADY_EXISTS;
+    }
+  }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    remove()                                                      **
+  **                                                                        **
+  ** Description: Takes key parameter.Removes the corresponding entry from  **
+  **              the map. If key does not exists returns error             **
+  **                                                                        **
+  ***************************************************************************/
+  map_rc_t remove(const keyT key) {
+    if (umap.empty()) {
+      return MAP_EMPTY;
+    }
+
+    std::lock_guard<std::mutex> lock(mutex_obj);
+    if (umap.erase(key)) {
+      return MAP_OK;
+    } else {
+      return MAP_KEY_NOT_EXISTS;
+    }
+  }
+};
+
+// Map Declarations:
+// Map- Key: uint64_t , Data: uint64_t
+typedef map_ts_s<uint64_t, uint64_t> map_uint64_ts_t;
```

Comment:
I think it would be good to start using a definitive naming for the maps, could be `map_<keyType>_<valueType>_ts_t`, in this case (`map_uint64_uint64_ts_t`)

Reply:
This comment has been addressed.